### PR TITLE
update: Added queue

### PR DESCRIPTION
**Pull request by Adman**
- print file by file from the queue
- TODO: we do need to somehow detect that
  object that was printed was safely removed from
  the plate and then start printing another one

Signed-off-by: Adrian Matejov a.matejov@centrum.sk

### CODE REVIEW

**Comment by Adman** [comment]
to @mrshu

File: printbot.py [hunk 1]
```diff
@@ -143,6 +149,17 @@ def __init__(self, opts=None):
...
 
+        self.queue = []
+        self.printing = False
+
+    def notify_user(self, entry, pos):
+        """Send notification to the user about his current queue position"""
+        msg = "Your file {0} is {1}/{2} in the queue".format(
+                                entry.filename,
+                                pos,
+                                str(len(self.queue)))
```
Comment (mrshu) [inline printbot.py]: @Adman this is not necessary, `format` can do the `str` for you.

**Comment by mrshu** [comment]
@Adman One last small nit, when testing this I noticed that once that this happened:

```
me: sent file X
PrintBot: File X is 1/1 in queue
PrintBot: Started printing X
me: sent file Y
PrintBot: File Y is 1/1 in queue

PrintBot: Printed X
PrintBot: Started printing Y
PrintBot: Printed Y
```

While it is technically correct, I would much rather count the currently printed file in the queue. `1/1` would suggest that it is already my turn and that my file will be printed right away. That will not happen for obvious reasons (to us, not necessarily the user) and might suggest that something is wrong where it is actually now.

**Comment by mrshu** [comment]
@Adman cool, this works, thanks!

**Pull request merged**